### PR TITLE
removed global package cache as it was causing issues

### PR DESCRIPTION
**Pull request by nouseforaname**
### What is this change about?

[GH Fixes: #1478, #1300, #353]

[finishes #3494307298](https://www.pivotaltracker.com/story/show/176753258)

### What tests have you run against this PR?
```
cd src/
bundle install 
bundle exec rake spec:unit
```

### How should this change be described in bosh release notes?
Remove obsolete & problematic code path. Obsolete config value `compiled_package_cache`

### Does this PR introduce a breaking change?

It makes a property on the director spec obsolete and removes support for it

### CODE REVIEW

File: src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb [hunk 1]
```diff
@@ -906,16 +905,12 @@ def compile_package_with_url_encrypt_stub(args)
       before do
```
Comment (beyhan) [inline src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb:905]: It looks like that we should remove the `describe 'the global blobstore'` completely here. The tests doesn't make much sense or do I miss something here?

**Review by beyhan — CHANGES_REQUESTED** [review]

File: src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb (hunk 1 above)
Comment (nouseforaname) [reply src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb]: you're right. removed it.

**Review by beyhan — APPROVED** [review]

**Review by rkoster — APPROVED** [review]
LGTM

**Comment by rkoster** [comment]
Thanks @nouseforaname

**Pull request merged**